raiden/tests/fixtures/variables.py

```diff
@@ -319,3 +325,30 @@ def environment_type():
 def unrecoverable_error_should_crash():
     """For testing an UnrecoverableError should crash"""
     return True
+
+
+@pytest.fixture
+def transport(request):
+    """ 'all' replaced by parametrize in conftest.pytest_generate_tests """
+    return request.config.getoption('transport')
+
+
+@pytest.fixture
+def transport_protocol(transport):
+    return TransportProtocol(transport)
+
+
+@pytest.fixture
+def skip_if_not_udp(request):
```

Comment:
Nitpick: why don't `skip_if_not_udp` and `skip_if_not_matrix` use the new `transport_protocol` fixture?

Reply:
They probably can now. `transport_protocol` etc. used to be defined only for the integration tests. I'll update it.

Reply:
Actually, thinking more about this we should leave those as is. Otherwise we would have a circular dependency `pytest_generate_tests -> skip_if_* -> transport_protocol -> transport -> pytest_generate_tests`